src/Components/Components/src/Forms/EditForm.cs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Components.RenderTree;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    /// <summary>
+    /// Renders a form element that cascades an <see cref="EditContext"/> to descendants.
+    /// </summary>
+    public class EditForm : ComponentBase
+    {
+        private readonly Func<Task> _handleSubmitDelegate; // Cache to avoid per-render allocations
+
+        private EditContext _fixedEditContext;
+
+        /// <summary>
+        /// Constructs an instance of <see cref="EditForm"/>.
+        /// </summary>
+        public EditForm()
+        {
+            _handleSubmitDelegate = HandleSubmitAsync;
+        }
+
+        /// <summary>
+        /// Supplies the edit context explicitly. If using this parameter, do not
+        /// also supply <see cref="Model"/>, since the model value will be taken
+        /// from the <see cref="EditContext.Model"/> property.
+        /// </summary>
+        [Parameter] EditContext EditContext { get; set; }
+
+        /// <summary>
+        /// Specifies the top-level model object for the form. An edit context will
+        /// be constructed for this model. If using this parameter, do not also supply
+        /// a value for <see cref="EditContext"/>.
+        /// </summary>
+        [Parameter] object Model { get; set; }
+
+        /// <summary>
+        /// Specifies the content to be rendered inside this <see cref="EditForm"/>.
+        /// </summary>
+        [Parameter] RenderFragment<EditContext> ChildContent { get; set; }
+
+        /// <summary>
+        /// A callback that will be invoked when the form is submitted.
+        ///
+        /// If using this parameter, you are responsible for triggering any validation
+        /// manually, e.g., by calling <see cref="EditContext.Validate"/>.
+        /// </summary>
+        [Parameter] Func<EditContext, Task> OnSubmit { get; set; }
+
+        /// <summary>
+        /// A callback that will be invoked when the form is submitted and the
+        /// <see cref="EditContext"/> is determined to be valid.
+        /// </summary>
+        [Parameter] Func<EditContext, Task> OnValidSubmit { get; set; }
+
+        /// <summary>
+        /// A callback that will be invoked when the form is submitted and the
+        /// <see cref="EditContext"/> is determined to be invalid.
+        /// </summary>
+        [Parameter] Func<EditContext, Task> OnInvalidSubmit { get; set; }
```

Comment:
I'm assuming that these become `EventCallback<EditContext>` 😀

Reply:
Yes! Done.